src/coreclr/vm/ceeload.cpp

```diff
@@ -4778,6 +4810,30 @@ VASigCookie *Module::GetVASigCookie(Signature vaSignature)
             pCookie->pNDirectILStub = NULL;
             pCookie->sizeOfArgs = sizeOfArgs;
             pCookie->signature = vaSignature;
+            pCookie->classInstCount = classInstCount;
+            pCookie->methodInstCount = methodInstCount;
+            
+            if (classInstCount != 0)
+            {
+                DWORD infoSize;
+                DWORD allocSize = DictionaryLayout::GetDictionarySizeFromLayout(classInstCount, NULL, &infoSize);
```

Comment:
This is not a generic dictionary. It is not appropriate to call `DictionaryLayout::GetDictionarySizeFromLayout` to multiple the two numbers. Multiply the two numbers inline instead.